### DISCUSSION TITLE
mousewheel support for older FMX 10.x

### DISCUSSION THREAD

**Discussion by manyplay** [DelphiWorlds/Kastri#223] [Q&A]
hi, 
thank you for your ideas and work!
i owned a license for that version and use several hmdi sticks (A 5.x up to 12) to use them as a sat2ip distributor with a small app i wrote.
(manyplay.de) - but for years i do miss really a wheel support by embarcadero - until the latest Version i knew that there is nothing happend.
i found your code tipps on [https://stackoverflow.com/questions/65511422/delphi-android-cant-capture-mouse-wheel-events](url)
and ask for a support on an older FMX version:
`WindowHandleToPlatform(Handle).FormLayout.addView(FGenericView);`
was not working on my version

**Comment by DelphiWorlds** [comment]
> and ask for a support on an older FMX version:
> `WindowHandleToPlatform(Handle).FormLayout.addView(FGenericView);`
> was not working on my version

What do you mean by "not working"? The code should compile, so what exactly is the trouble?

**Comment by manyplay** [comment]
no trouble - but if i try to compile it with 10.1.2 following is undeclared in the compiler message:
`WindowHandleToPlatform(Handle).**FormLayout**.addView(FGenericView)`

**Reply by manyplay** [reply]
trying in 10.4.2 works but i can not get any wheel scroll (pos/neg) value in the JMotionEvent´s - only axis/press values and some infos

hoped that i get a MouseWheelListener similiar to this:
[https://docs.oracle.com/javase/tutorial/uiswing/events/mousewheellistener.html](url)

with your code...

**Comment by DelphiWorlds** [comment]
Can not help you with earlier than Delphi 10.4.2.

> i can not get any wheel scroll (pos/neg) value in the JMotionEvent´s - only axis/press values and some infos

The AXIS values are exactly what you need to examine, as per the docs, e.g [AXIS_VSCROLL](https://developer.android.com/reference/android/view/MotionEvent#AXIS_VSCROLL)

There's a Java example [here](https://stackoverflow.com/a/33086042/3164070), but it should be easy enough to convert to Delphi.

**Reply by manyplay** [reply]
thanks for the informations!

its working - $9 was the axis code for vscroll value
